airflow/contrib/hooks/spark_submit_hook.py

```diff
@@ -255,12 +254,11 @@ def _process_log(self, itr):
 
     def on_kill(self):
         if self._sp and self._sp.poll() is None:
-            logging.info('Sending kill signal to spark-submit')
-            self.sp.kill()
+            logging.info('Sending kill signal to %s' % self._connection['spark_binary'])
+            self._sp.kill()
 
             if self._yarn_application_id:
                 logging.info('Killing application on YARN')
```

Comment:
Well, no, since logging that the yarn app is killed is not meant to be displayed  in a case of a failure of the the killcmd.

maybe with something like : 

```python
try:
    p = subprocess.check_call(kill_cmd)
    logging.info("YARN killed with return code: {0}".format(p.returncode))
except subprocess.CalledProcessError as e
   logging.error("Something went wrong kill the YARN app")
   logging.error("return code is : {}".format(e.returncode)
   logging.error("output is : {}".format(e.output)
```

Trapping the killcmd in a case of a failure is maybe not a good idea ? 

Anyone ?